plugins/inputs/smart/smart.go

```diff
@@ -62,6 +52,55 @@ var (
 		"194": "temp_c",
 		"199": "udma_crc_errors",
 	}
+
+	sasNvmeAttributes = map[string]struct {
+		ID    string
+		Name  string
+		Parse func(fields, deviceFields map[string]interface{}, str string) error
+	}{
```

Comment:
Why is this a map instead of a struct?  It seems it doesn't need any dynamic items?

Reply:
@docmerlin so it goes:
1. Generically match anything in the form `(Some Attribute): (1234)`
2. Lookup "Some Attribute" in `sasNvmeAttributes`
3. If match use Name field of attribute (as this differs from match in smartctl output) as name in metric point.
4. Use ID in point if set
5. Parse value as integer (which may contain a ","). If override parse function is supplied, use that instead (useful for temperatures which also populate deviceFields and percentages).

Perhaps I am missing something. But I wanted to avoid having to search through every possible attribute name that could be in the line. In favor of generically pulling out anything `key: value` and looking up the key in a set of known attribute keys.